.travis.yml

```diff
@@ -1,48 +1,120 @@
-env:
-   - CABALVER=1.18 GHCVER=7.4.1 # Debian 7
-   - CABALVER=1.20 GHCVER=7.6.3 # Debian 8
-   - CABALVER=1.18 GHCVER=7.8.4
-   - CABALVER=1.22 GHCVER=7.10.1
-   - CABALVER=1.24 GHCVER=8.0.1
-   - CABALVER=1.24 GHCVER=8.2.1
-   - CABALVER=head GHCVER=head
+# This Travis job script has been generated by a script via
+#
+#   runghc make_travis_yml_2.hs 'list-tries.cabal'
+#
+# For more information, see https://github.com/hvr/multi-ghc-travis
+#
+language: c
+sudo: false
+
+git:
+  submodules: false  # whether to recursively clone submodules
+
+cache:
+  directories:
+    - $HOME/.cabal/packages
+    - $HOME/.cabal/store
+
+before_cache:
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/build-reports.log
+  # remove files that are regenerated by 'cabal update'
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/00-index.*
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/*.json
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.cache
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar
+  - rm -fv $HOME/.cabal/packages/hackage.haskell.org/01-index.tar.idx
+
+  - rm -rfv $HOME/.cabal/packages/head.hackage
 
 matrix:
-   allow_failures:
-      - env: GHCVER=head CABALVER=head
+  include:
+    - compiler: "ghc-7.0.4"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-head,ghc-7.0.4], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.2.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-head,ghc-7.2.2], sources: [hvr-ghc]}}
+    - compiler: "ghc-7.4.2"
+    # env: TEST=--disable-tests BENCH=--disable-benchmarks
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-head,ghc-7.4.2], sources: [hvr-ghc]}}
```

Comment:
`cabal-install` is not shipped with GHC, it's perfectly fine to use new `cabal-install` versions with old GHCs, in fact I'd say it's recommended. I'll see if the change you want is easy to do, otherwise I'll just drop `.travis.yml` changes.

Reply:
Right, I was thinking of Cabal-the-library. In any case I think my original intention was again based on the Debian distros: I looked up what their APT repositories offer out of the box and tested with that. People might not know about such recommendations, so I think this way is safer.